File: Java/src/main/java/com/gildedrose/item/AgedBrie.java
/* (C)2022 */
package com.gildedrose.item;

public class AgedBrie extends Item {

    public static final String AGED_BRIE = "Aged Brie";

    public AgedBrie(int sellIn, int quality) {
        super(AGED_BRIE, sellIn, quality);
    }

    @Override
    public void updateQuality() {
        if (quality < 50) {
            if (sellIn <= 0) {
                quality += 2;
            } else {
                quality += 1;
            }
        }
    }
}

File: csharpcore/GildedRose/Models/AgedBrie.cs
using GildedRose.Abstraction;
using GildedRoseKata;

namespace GildedRose.Models
{
    public class AgedBrie : Item, ICustomMethod
    {
        public int SellDaysGone { get; set; }

        public void UpdateQuality() {
            if (this.Quality < 51)
            {
                this.Quality++;
            }
        }

        public void UpdateSellIn()
        {
            if (this.SellIn > 0)
                this.SellIn--;
        }
    }
}